src/test/java/org/openrewrite/java/migrate/RemovedToolProviderConstructorTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.DocumentExample;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+import static org.openrewrite.java.Assertions.java;
+
+class RemovedToolProviderConstructorTest  implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+            spec.expectedCyclesThatMakeChanges(2).recipe(new RemovedToolProviderConstructor());
+    }
+
+    @DocumentExample
+    @Test
+    void moveToStaticTest() {
+            rewriteRun(
+              //language=java
+              java(
+                """
+                  package com.test;
+                   
+                  import javax.tools.ToolProvider;
+                   
+                  public class RemovedToolProviderConstructorApp {
+                   
+                       public void test() throws Exception {
+                           ToolProvider tp = null;
+                           tp.getSystemJavaCompiler();     
+                           tp.getSystemDocumentationTool();
+                           tp.getSystemToolClassLoader();  
+                           System.out.println(ToolProvider.getSystemJavaCompiler());      
+                       }
+                  }          
+                  """,
+                """
+                  package com.test;
+                   
+                  import javax.tools.ToolProvider;
+                   
+                  public class RemovedToolProviderConstructorApp {
+                   
+                       public void test() throws Exception {
+                           ToolProvider tp = null;
+                           ToolProvider.getSystemJavaCompiler();     
```

Comment:
We need to broaden this test case to capture more ways these calls can be make that better illustrate real world scenarios(for example take a look at how `ToolProvider.getSystemJavaCompiler()` is called here http://www.java2s.com/example/java-api/javax/tools/toolprovider/getsystemjavacompiler-0-8.html). I think some of the logic in your recipe can be stripped but we need better test coverage to confirm this. Some examples of

Reply:
added in the next commit